inc/cli/class-commands.php

```diff
@@ -0,0 +1,295 @@
+				'paged'               => $paged,
+				'post_status'         => 'any',
+				'ignore_sticky_posts' => true,
+				'suppress_filters'    => false,
+			] );
+
+			// Exit early if there are no more posts to avoid a final sleep call.
+			if ( empty( $posts ) ) {
+				break;
+			}
+
+			foreach ( $posts as $post ) {
+				$authorship_authors = \Authorship\get_authors( $post );
+
+				if ( ! empty( $authorship_authors ) && ! $overwrite ) {
+					// skip posts that already have authorship data.
+					WP_CLI::line( 'Skipping post that already has Authorship data' );
+					continue;
+				}
+
+				// check for PPA data.
+				$ppa_terms = wp_get_object_terms( $post->ID, [ 'author' ] );
+
+				// if there are no Publish Press Authors data then there's
+				// nothing to migrate, skip!
+				if ( empty( $ppa_terms ) ) {
+					WP_CLI::line( 'Skipping post with no PPA data' );
+					continue;
+				}
+
+				// If PPA is deactivated the terms can be an error object.
+				// Usually invalid taxonomy, lets catch and report this.
+				if ( is_wp_error( $ppa_terms ) ) {
+					WP_CLI::error( 'There was an error fetching the Publishpress Author data, is the plugin activated?', false );
+					WP_CLI::error( $ppa_terms );
+					exit;
+				}
+
+				/**
+				 * We're going to loop through all the Publishpress authors,
+				 * and get a list of user IDs to set in Authorship at the end.
+				 * If no user exists for the PPA guest author we create one.
+				 */
+				$new_authors = [];
+				foreach ( $ppa_terms as $ppa_author ) {
+					// add this user ID to the list of authors.
+					$user_id = $this->get_ppa_user_id( $ppa_author, $dry_run );
+
+					// on dry runs the ID might be -1 if a user needed to be created.
+					if ( $user_id !== -1 ) {
+						$new_authors[] = $user_id;
+					}
+				}
+
+				if ( empty( $new_authors ) ) {
+					WP_CLI::line( 'Skipping post that has no multi-author data' );
+					continue;
+				}
+
+				if ( ! $dry_run ) {
+					\Authorship\set_authors( $post, $new_authors );
+				}
+
+				$count++;
+			}//end foreach
+
+			// Avoid memory exhaustion issues.
+			$this->reset_local_object_cache();
+
+			// Pause for a moment to let the database catch up.
+			WP_CLI::line( sprintf( 'Processed %d posts, pausing for a breath...', $count ) );
+			sleep( 2 );
+
+			$paged++;
+		} while ( count( $posts ) );
+
+		if ( true === $dry_run ) {
+			WP_CLI::success( sprintf( '%d posts would have had Authorship data added if this was not a dry run.', $count ) );
+		} else {
+			WP_CLI::success( sprintf( '%d posts have had Authorship data added.', $count ) );
+		}
+	}
+
+	/**
+	 * Takes a PublishPress Author term and returns a user ID
+	 * for Authorship.
+	 *
+	 * @param WP_Term $ppa_author The term for the PPA author
+	 * @param boolean $dry_run If true no users will be created if they are missing
+	 *
+	 * @return integer a user ID for this term, or -1 if not resolvable
+	 */
+	private function get_ppa_user_id( WP_Term $ppa_author, bool $dry_run ) : int {
```

Comment:
It's unexpected that a `get_*` function creates anything, let alone users. I'd recommend to default to false, update PHPDoc and rename the function to something else such as `ensure_ppa_user` as a creation function would be expected to return an id of the created object?

```suggestion
	private function get_ppa_user_id( WP_Term $ppa_author, bool $create_users = false ) : int {
```